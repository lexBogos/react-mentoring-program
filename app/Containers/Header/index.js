import React from "react";
import SearchPanel from "../../Components/SearchPanel";
import AddMovie from "../../Components/AddMovie";
import MovieDetails from '../../Components/MovieDetails';
import "./index.scss";

const addMovieModalFields = (hook, list, setActive) => {
  const movieObj = {
    title: "",
    realiseDate: "",
    movieUrl: "",
    genre: "sci-fi",
    overview: "",
    runTime: "",
  };

  return (
    <div className="addMovie">
      <h3>Add Movie</h3>
      <div>
        <span>TITLE</span>
        <input
          onChange={(event) => {
            movieObj.title = event.target.value;
          }}
        ></input>
      </div>
      <div>
        <span>RELEASE DATE</span>
        <input
          onChange={(event) => {
            movieObj.realiseDate = event.target.value;
          }}
        ></input>
      </div>
      <div>
        <span>MOVIE URL</span>
        <input
          onChange={(event) => {
            movieObj.movieUrl = event.target.value;
          }}
        ></input>
      </div>
      <div>
        <span>GENRE</span>
        <select
          name="genre"
          id="genre"
          onChange={(event) => {
            movieObj.genre = event.target.value;
          }}
        >
          <option value="sci-fi">Sci-Fi</option>
          <option value="thriller">Thriller</option>
          <option value="action">Action</option>
        </select>
      </div>
      <div>
        <span>OVERVIEW</span>
        <input
          onChange={(event) => {
            movieObj.overview = event.target.value;
          }}
        ></input>
      </div>
      <div>
        <span>RUNTIME</span>
        <input
          onChange={(event) => {
            movieObj.runTime = event.target.value;
          }}
        ></input>
      </div>
      <div className="modalButton">
        <button
          onClick={() => {
            setActive(false);
          }}
        >
          RESET
        </button>
        <button
          onClick={() => {
            hook([...list, { ...movieObj, id: list.length }]);
            setActive(false);
          }}
        >
          SUBMIT
        </button>
      </div>
    </div>
  );
};

const Header = ({ setModalActive, setModalFields, choosenMovie, setChoosenMovie, movieList }) => {

  return (
    <div className="headerContainer">
      <div className="headerImage"></div>
      {choosenMovie === -1 ? (
        <>
          <h1 className="title">Find your movie</h1>
          <AddMovie
            setModalFields={setModalFields}
            func={addMovieModalFields}
            setModalActive={setModalActive}
          />
          <div className="searchContainer">
            <SearchPanel />
          </div>
        </>
      ) : (
        <>
            <MovieDetails choosenMovie={choosenMovie} setChoosenMovie={setChoosenMovie} movieList={movieList}/>
        </>
      )}
    </div>
  );
};

export default Header;