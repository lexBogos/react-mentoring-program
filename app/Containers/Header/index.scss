.headerContainer{
    background-image: url('https://assets.nflxext.com/ffe/siteui/vlv3/e178a4e7-4f52-4661-b2ae-41efa25dca7c/eb618394-24cd-43ae-8799-c1a76a7bcbfc/BY-en-20210222-popsignuptwoweeks-perspective_alpha_website_large.jpg');
    height: 500px;
    font-size:30px;
    font-weight: 400;
    color: #fff;
}

.searchContainer{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}


.btn {
    border-radius: 5%;
    color: #fff;
    cursor: pointer;
    font-size:16px;
    font-weight: 400;
    line-height: 45px;
    margin: 0 0 2em;
    max-width: 160px; 
    position: relative;
    text-decoration: none;
    text-transform: uppercase;
    width: 100%; 
    background-color: rgb(246,82,97);
    margin-right: 20%;
    
  }

.btn:active {
    background-color: rgb(246,82,97);
    box-shadow: 0 5px #666;
    transform: translateY(4px);
  }