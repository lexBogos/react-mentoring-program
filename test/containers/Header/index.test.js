import React from 'react';
import Header from '../../../app/containers/Header/index';
import '@testing-library/jest-dom/extend-expect';
import { createStore } from "redux";
import reducer from "../../../app/reducers/index";
import { Provider } from "react-redux";

import { render as rtlRender } from '@testing-library/react';

function render(
    ui,
    {
      initialState,
      store = createStore(reducer, initialState),
      ...renderOptions
    } = {}
  ) {
    function Wrapper({ children }) {
      return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
  }

jest.mock('react-router-dom', () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: '/',
      search: '',
      hash: '',
      state: null,
      key: '5nvxpbdafa',
    }),
    useHistory: jest.fn().mockReturnValue({
        pathname: '/',
        search: '',
        hash: '',
        state: null,
        key: '5nvxpbdafa',
      }),
}));

describe('Header component', () => {
    it('render Header component toMatchSnapshot', () => {
        const { asFragment } = render((<Header/>));
        expect(asFragment(<Header/>)).toMatchSnapshot();
    })
})