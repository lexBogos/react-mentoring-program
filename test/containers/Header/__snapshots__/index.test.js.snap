// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render Header component toMatchSnapshot 1`] = `
<DocumentFragment>
  <div
    class="headerContainer"
  >
    <div
      class="headerImage"
    />
    <h1
      class="title"
    >
      Find your movie
    </h1>
    <button
      class="addMovieButton"
    >
      + ADD MOVIE
    </button>
    <div
      class="searchContainer"
    >
      <input
        class="searchInput"
        placeholder="What do you want to see ?"
      />
      <button
        class="btn btn-1"
      >
        SEARCH
      </button>
    </div>
  </div>
</DocumentFragment>
`;
